content/tests/components/pages/OverviewPage.test.tsx

```diff
@@ -100,4 +102,93 @@ describe("OverviewPage", () => {
 
     expect(NormandyAPI.prototype.disableRecipe).toBeCalled();
   });
+
+  it("should display need to pause recipes", async () => {
+    const recipes = recipeFactory.buildCount(3);
+    const enrollmentPaused = [10000, 2, 5];
+    const hundredDaysFuture = new Date();
+    hundredDaysFuture.setDate(hundredDaysFuture.getDate() + 100);
+    const experiments = experimenterResponseFactory.buildMany(
+      recipes.map((r, i) => ({
+        normandy_id: r.id,
+        proposed_enrollment: enrollmentPaused[i],
+        end_date: hundredDaysFuture.getTime(),
+      })),
+    );
+    jest
+      .spyOn(NormandyAPI.prototype, "fetchApprovalRequests")
+      .mockImplementation(() => Promise.resolve([]));
+
+    const mockFetchRecipe = jest
+      .fn()
+      .mockImplementation((i): RecipeV3 => recipes[i % 3]);
+
+    jest
+      .spyOn(NormandyAPI.prototype, "fetchRecipe")
+      .mockImplementation(() =>
+        Promise.resolve(mockFetchRecipe(mockFetchRecipe.mock.calls.length)),
+      );
```

Comment:
is this better than having 6 `mockReturnValueOnce`s or is this still considered clunky?

Reply:
Huh. I hadn't thought about using the call length itself. That seems clever. +1 to this.